src/transformers/generation/logits_process.py

```diff
@@ -1480,13 +1483,17 @@ class InfNanRemoveLogitsProcessor(LogitsProcessor):
     """
 
     @add_start_docstrings(LOGITS_PROCESSOR_INPUTS_DOCSTRING)
-    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> torch.FloatTensor:
+    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, cur_len: int) -> torch.FloatTensor:
         # set all nan values to 0.0
-        scores[scores != scores] = 0.0
+        zeros = torch.zeros_like(scores)
+        scores = scores.masked_scatter(scores != scores, zeros)
+
+        maxs = torch.ones_like(scores) * torch.finfo(scores.dtype).max
+        mins = torch.ones_like(scores) * torch.finfo(scores.dtype).min
 
         # set all +/-inf values to max/min possible value
-        scores[scores == float("inf")] = torch.finfo(scores.dtype).max
-        scores[scores == float("-inf")] = torch.finfo(scores.dtype).min
+        scores = scores.masked_scatter(scores == float("inf"), maxs)
+        scores = scores.masked_scatter(scores == -float("inf"), mins)
```

Comment:
nit: we don't need to create `maxs`/`mins` with `scores = torch.where(scores == float("inf"), torch.finfo(scores.dtype).max, scores)`, which is (IMO) easier to read